lib/src/disposable/disposable.dart

```diff
@@ -300,8 +300,20 @@ class Disposable implements _Disposable, DisposableManagerV3 {
   @override
   void manageStreamController(StreamController controller) {
     _throwOnInvalidCall('manageStreamController', 'controller', controller);
+    // If a single-subscription stream has a subscription and that
+    // subscription is subsequently canceled, the `done` future will
+    // complete, but there is no other way for us to tell that this
+    // is what has happened. If we then listen to the stream (since
+    // closing a stream that was never listen-to throws) we'll get
```

Comment:
I was under the impression that closing a single-subscription StreamController that had not been listened to would just never complete, not throw